spec.md

```diff
@@ -2821,6 +2981,7 @@ All other strings, such as elements of a list of strings, are double-quoted.
 str(1)                          # '1'
 str("x")                        # 'x'
 str([1, "x"])                   # '[1, "x"]'
+str(0.0)                        # '0.0'        (formatted as if by "%g")
```

Comment:
Mind the issue where Python's `%g` doesn't force the number to end in `.0` when equivalent to an integer. (If that's intentional, document it with `%g`.)

Reply:
Noted

The compact form `%g` is also used by `str(float)`. Its result                                       
uses the least precision required to accurately represent the value,                                 
omits unnecessary trailing zeros, and always contains a period                                       
and/or an exponent and thus unambiguously denotes a `float`, not an `int`.

Reply:
> omits unnecessary trailing zeros, and always contains a period and/or an exponent

Nit: This implies that it prints `5.` instead of `5.0`.

Reply:
(Forgot about this comment before I opened a new one, ah well.)